### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by Sky322**

### CODE REVIEW

**Comment by Sky322** [comment]
added a win condition which is just a orange square that changes location and fail condition is the fireballs. Also controls are either WASD or arrow keys

File: source/PlayState.hx [hunk 1]
```diff
@@ -1,33 +1,44 @@
 package;
 
+import js.lib.webassembly.Global;
```
Comment (SamBumgardner) [inline source/PlayState.hx]: I don't think this is used anywhere, and including it prevents the game from building to the neko target via `lime test neko`

I'd like this removed before merging, please.

File: source/PlayState.hx [hunk 2]
```diff
@@ -62,16 +73,18 @@ class PlayState extends FlxState
...
 
-		// Wrap various objects if gone offscreen.
+		FlxG.overlap(hero, goal, heroGoalOverlap);
+
 		screenWrapObject(hero);
 		for (fireball in fireballs) {
 			screenWrapObject(fireball);
+
+		ifEnding();
+		ifWon();
```
Comment (SamBumgardner) [inline source/PlayState.hx:87]: I don't think you meant to put these function calls inside your fireball object-wrapping loop. As-is, they get called way more often than they need to.

**Review by SamBumgardner — CHANGES_REQUESTED** [review]
There's one issue that prevents neko builds, so I'd like that fixed before merging. I leave the other issue to fix at your discretion.

Other than that, it looked good. I liked dodging between fireballs quite a bit! 
Will check back in after it's fixed to give final approval.

**Pull request merged**